narwhals/_spark_like/expr.py

```diff
@@ -102,7 +102,9 @@ def func(df: SparkLikeLazyFrame) -> list[Column]:
             for _input in inputs:
                 input_col_name = get_column_name(df, _input)
                 column_result = call(_input, **_kwargs)
-                if not returns_scalar:
+                if not returns_scalar or (
+                    (self._depth == 0) and (self._function_name == "all")
```

Comment:
thanks for looking into this!

I think this might be too specific and that we'll need a more generic solution - will look into it and let you know